pkg/domain/infra/tunnel/helpers.go

```diff
@@ -14,15 +14,15 @@ import (
 // FIXME: the `ignore` parameter is very likely wrong here as it should rather
 //        be used on *errors* from operations such as remove.
 func getContainersByContext(contextWithConnection context.Context, all, ignore bool, namesOrIDs []string) ([]entities.ListContainer, error) {
-	ctrs, _, err := getContainersAndInputByContext(contextWithConnection, all, ignore, namesOrIDs)
+	ctrs, _, err := getContainersAndInputByContext(contextWithConnection, all, ignore, namesOrIDs, nil)
 	return ctrs, err
 }
 
-func getContainersAndInputByContext(contextWithConnection context.Context, all, ignore bool, namesOrIDs []string) ([]entities.ListContainer, []string, error) {
+func getContainersAndInputByContext(contextWithConnection context.Context, all, ignore bool, namesOrIDs []string, filters map[string][]string) ([]entities.ListContainer, []string, error) {
 	if all && len(namesOrIDs) > 0 {
 		return nil, nil, errors.New("cannot look up containers and all")
 	}
-	options := new(containers.ListOptions).WithAll(true).WithSync(true)
+	options := new(containers.ListOptions).WithAll(true).WithSync(true).WithFilters(filters)
```

Comment:
I think we're filtering on the client side again.. We should take another look at this, Stop should be filtered within stop, and not within list.

Reply:
I believe we are filtering on the server side because ithe filtered containers are being returned from the ContainerStop() call on line 66 in pkg/api/handlers/compat/containers_stop.go. This stops the filtered containers but does not lead to output of the filtered containers on the client side. Without these changes, all container IDs are printed no matter what is filtered

Reply:
if the .WithFilters(filters) is removed, the correct containers are filtered and stopped, but all IDs are printed, not the filtered IDs

Reply:
Ohhhh okay I see what's going on. We're using list to call the filters and then hitting the stop API one by one, since the Stop API mandates an ID. This means we don't actually have to generate a filter for the Stop rest API, and we use List to get the name and IDs of the containers to stop, and then make multiple calls to the Stop endpoint. Okay, my bad. I think we need to remove the compat handlers for the Stop endpoint for filters then, since filters will never actually make it through the wire on the Stop side, but rather be called from the list side.